renderservice/render/daemon/ResourceGenerator.py

```diff
@@ -0,0 +1,168 @@
+        """Return a response containing a generated PDF resource.
+
+        Args:
+            task: A dicitionary of values specifying the task.
+                Must have:
+                  - resource_slug
+                  - resource_name
+                  - resource_view
+                  - header_text (optional)
+                  - paper_size
+                  - copies
+                  - url
+
+        Returns:
+            Tuple of filename and PDF file of generated resource.
+        """
+        if task.get("resource_slug", None) is None:
+            raise TaskError("Task must specify the resource slug.")
+        if task.get("resource_name", None) is None:
+            raise TaskError("Task must specify the resource name.")
+        if task.get("resource_view", None) is None:
+            raise TaskError("Task must specify the resource view.")
+        if task.get("paper_size", None) is None:
+            raise TaskError("Task must specify paper size.")
+        if task.get("copies", None) is None:
+            raise TaskError("Task must specify number of copies.")
+        if task.get("url", None) is None:
+            raise TaskError("Task must specify the url.")
+
+        resource_generator = self.import_resource_module(task["resource_view"])
+
+        for option, values in resource_generator.valid_options().items():
+            if option not in task.keys():
+                raise TaskError("Task is missing value for {}.".format(option))
+            if task[option] not in values:
+                raise TaskError("Value ({}) for option {} is not in: {}.".format(task[option], option, values))
+
+        context = dict()
+        context["resource_name"] = task["resource_name"]
+        context["header_text"] = task.get("header_text", "")
+        context["paper_size"] = task["paper_size"]
+        context["url"] = task["url"]
+
+        context["all_data"] = []
+        for copy in range(0, task["copies"]):
+            context["all_data"].append(
+                self.generate_resource(task, resource_generator)
+            )
+
+        filename = "{} ({}).pdf".format(task["resource_name"], resource_generator.subtitle(task))
+        context["filename"] = filename
+
+        template_filename = task.get("template", "base-resource-pdf.html")
+        css_filename = task.get("css", "css/print-resource-pdf.css")
+
+        template = self.template_environment.get_template(template_filename)
+        pdf_html = template.render(context)  # TODO: Future consider async
+        html = HTML(string=pdf_html, base_url=STATIC_DIRECTORY)
+        css_data = self.file_manager.load(css_filename).read()
+        css_string = css_data.decode("utf-8")
+        base_css = CSS(string=css_string)
+        return filename, html.write_pdf(stylesheets=[base_css])
+
+    def generate_resource(self, task, resource_generator):
+        """Retrieve page(s) for one copy of resource from resource generator.
+
+        Images are resized to size.
```

Comment:
what?

Reply:
This is what is was on the previous system. Changed now to "resized to fit page".